packages/components/src/components/horizontal-nav/horizontal-nav.component.ts

```diff
@@ -0,0 +1,196 @@
+import { property, query, state } from 'lit/decorators.js';
+import componentStyles from '../../styles/component.styles.js';
+import SynergyElement from '../../internal/synergy-element.js';
+import styles from './horizontal-nav.styles.js';
+import SynIcon from '../icon/icon.component.js';
+import SynNavItem from '../nav-item/nav-item.component.js';
+
+/**
+ * @summary The `<syn-horizontal-nav />` element provides a generic navigation bar that
+ * can be used to group multiple `<syn-nav-item />` elements together.
+ * It will group all `<syn-nav-item />`s that cannot be displayed into a custom priority menu
+ *
+ * @documentation https://synergy-design-system.github.io/?path=/docs/components-syn-horizontal-nav--docs
+ * @status stable
+ * @since 1.10
+ *
+ * @dependency syn-icon
+ * @dependency syn-nav-item
+ *
+ * @slot - The given navigation items. Must be `<syn-nav-item>` elements.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart nav-item-wrapper - The wrapper around the slotted `<syn-nav-item />` elements
+ * @csspart priority-menu - The wrapper around the priority menu
+ * @csspart priority-menu-label - The label for the priority menu
+ *
+ * @cssproperty --navigation-spacing - The amount of padding to use for the horizontal navigation.
+ */
+export default class SynHorizontalNav extends SynergyElement {
+  static styles: CSSResultGroup = [componentStyles, styles];
+
+  static dependencies = {
+    'syn-icon': SynIcon,
+    'syn-nav-item': SynNavItem,
+  };
+
+  /**
+   * Internal resize observer
+   */
+  private resizeObserver: ResizeObserver;
+
+  /**
+   * Reference to the rendered children slot
+   */
+  @query('slot') defaultSlot: HTMLSlotElement;
+
+  /**
+   * The wrapper that holds the horizontal navigation items
+   */
+  @query('.horizontal-nav-items') navItemWrapper: HTMLDivElement;
+
+  /**
+   * The priority menu wrapper
+   */
+  @query('.priority-menu') priorityMenu: HTMLDivElement;
+
+  /**
+   * Reference to the priority menu label
+   */
+  @query('.priority-menu__label') priorityMenuLabelElement: HTMLSpanElement;
+
+  /**
+   * The components priority menu label.
+   * This will be shown after the priority menu 3 dots link
+   */
+  @property({ attribute: 'priority-menu-label' }) priorityMenuLabel = 'Menu';
+
+  /**
+   * The items that are currently not completely visible and moved to the priority menu
+   */
+  @state() protected priorityMenuItems: SynNavItem[] = [];
+
+  /**
+   * Get a list of all slotted `<syn-nav-item />` elements
+   */
+  private getSlottedNavItems() {
+    const slottedItems = Array.from(
+      this.defaultSlot.assignedElements({ flatten: true }),
+    ) as HTMLElement[];
+
+    return slottedItems.filter(i => i instanceof SynNavItem) as SynNavItem[];
+  }
+
+  /**
+   * Adjust the visual placement of the priority menu
+   * @param startPos The start position to calculate from
+   */
+  private adjustPriorityMenuPosition(startPos: number) {
```

Comment:
```suggestion
  private adjustPriorityMenuPosition(startPosition: number) {
```

We barely use abbreviations in code.